code/game/machinery/kitchen/cooking_machines/_appliance.dm

```diff
@@ -0,0 +1,652 @@
+/obj/item/weapon/reagent_containers/food/snacks/var/list/cooked = list()
+
+// Root type for cooking machines. See following files for specific implementations.
+/obj/machinery/appliance
+	name = "cooker"
+	desc = "You shouldn't be seeing this!"
+	icon = 'icons/obj/cooking_machines.dmi'
+	var/appliancetype = 0
+	density = 1
+	anchored = 1
+
+	use_power = 1
+	idle_power_usage = 5			// Power used when turned on, but not processing anything
+	active_power_usage = 1000		// Power used when turned on and actively cooking something
+
+
+	var/cooking_power  = 1
+	var/max_contents = 1			// Maximum number of things this appliance can simultaneously cook
+	var/on_icon						// Icon state used when cooking.
+	var/off_icon					// Icon state used when not cooking.
+	var/cooking						// Whether or not the machine is currently operating.
+	var/cook_type					// A string value used to track what kind of food this machine makes.
+	var/can_cook_mobs				// Whether or not this machine accepts grabbed mobs.
+	var/mobdamagetype = BRUTE		// Burn damage for cooking appliances, brute for cereal/candy
+	var/food_color					// Colour of resulting food item.
+	var/cooked_sound				// Sound played when cooking completes.
+	var/can_burn_food				// Can the object burn food that is left inside?
+	var/burn_chance = 10			// How likely is the food to burn?
+	var/list/cooking_objs = list()	// List of things being cooked
+
+	// If the machine has multiple output modes, define them here.
+	var/selected_option
+	var/list/output_options = list()
+	var/list/datum/recipe/available_recipes // List of the recipes this appliance could possibly make
+
+	var/container_type = null
+
+	var/combine_first = 0//If 1, this appliance will do combinaiton cooking before checking recipes
+
+/obj/machinery/appliance/initialize()
+	..()
+	if(output_options.len)
+		verbs += /obj/machinery/appliance/proc/choose_output
+
+	if (!available_recipes)
+		available_recipes = new
+
+	for (var/type in subtypesof(/datum/recipe))
+		var/datum/recipe/test = new type
+		if ((appliancetype & test.appliance))
+			available_recipes += test
+		else
+			qdel(test)
+
+/obj/machinery/appliance/Destroy()
+	for (var/a in cooking_objs)
+		var/datum/cooking_item/CI = a
+		qdel(CI.container)//Food is fragile, it probably doesnt survive the destruction of the machine
+		cooking_objs -= CI
+		qdel(CI)
+	return ..()
+
+/obj/machinery/appliance/examine(var/mob/user)
+	..()
+	if(Adjacent(usr))
+		list_contents(user)
+		return 1
+
+/obj/machinery/appliance/proc/list_contents(var/mob/user)
+	if (cooking_objs.len)
+		var/string = "Contains..."
+		for (var/a in cooking_objs)
+			var/datum/cooking_item/CI = a
+			string += "-\a [CI.container.label(null, CI.combine_target)], [report_progress(CI)]</br>"
+		usr << string
+	else
+		usr << span("notice","It is empty.")
+
+/obj/machinery/appliance/proc/report_progress(var/datum/cooking_item/CI)
+	if (!CI || !CI.max_cookwork)
+		return null
+
+	if (!CI.cookwork)
+		return "It is cold."
```

Comment:
well its still correct for the cereal maker.

As for candy, i dunno, but im leaning towards heat being involved somehow